jetty-server/src/main/java/org/eclipse/jetty/server/HttpOutput.java

```diff
@@ -231,6 +229,8 @@ private void write(ByteBuffer content, boolean complete) throws IOException
 
     protected void write(ByteBuffer content, boolean complete, Callback callback)
     {
+        if (_firstByteTimeStamp == -1)
+            _firstByteTimeStamp = System.nanoTime();
```

Comment:
Should we check the getHttpConfiguration().getMinResponseDataRate() here and only set the timestamp if we need it. That way onFlushed only need check _firstByteTimeStamp and it knows it has a minResponseDataRate!